### ISSUE TITLE
Build fails with `Unexpected character #` in `node-fetch-native`

### ISSUE THREAD

**Issue by jmsfilipe** [nuxt/nuxt#10751]
### Versions

- nuxt-edge: 2.16.0-27720022.54e852f
- node: v16.17.0

### Reproduction

<details open>
<summary>Additional Details</summary>
<br>

</details>

### Steps to reproduce
- Use latest nuxt-edge version
- `nuxt.config.js` with `target: 'static'` and `ssr: true`
- Running nuxt build fails with
```
ERROR in ./node_modules/node-fetch-native/dist/index.cjs 69:2
Module parse failed: Unexpected character '#' (69:2)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|  */
| class BlobDataItem {
>   #path
|   #start
| 
 @ ./node_modules/node-fetch-native/lib/index.cjs 1:18-46
 @ ./.nuxt/server.js
 @ multi ./node_modules/@nuxt/components/lib/installComponents.js ./.nuxt/server.js
```

### What is Expected?
- Running nuxt build should be successful

### What is actually happening?
- Seems like the `node-fetch-native` dependency used to be just `node-fetch`
- This new dependency includes [private methods syntax](https://babeljs.io/docs/en/babel-plugin-proposal-private-methods) and webpack/eslint don't know how to deal with that? Maybe it needs to be transpiled?

**Comment by Billybobbonnet** [comment]
I have the exact same issue with regular Nuxt v2 (without the nuxt-edge package), same config.  I can't generate anymore.

**Comment by sanderjn** [comment]
I'm running into the same problem. Though I'm not using `target: 'static'` and `ssr: true`.

Local `dev` and `build` commands work just fine, but I'm getting this error just at Vercel.

**Comment by danielroe** [comment]
This seems an issue with webpack's support for class properties in node-fetch-native. We likely need to configure this in Nuxt 2.

cc: @pi0

**Comment by sanderjn** [comment]
Any workaround for now?

**Comment by Billybobbonnet** [comment]
> Any workaround for now?

Pin your Nuxt version to `"nuxt": "2.15.7"`

**Comment by jbmarflo** [comment]
I cant solve this issue yet, Can anybody solve this error? I'm tried with babel

**Comment by pimlie** [comment]
You could also add a alias from node-fetch-native back to node-fetch when using `v2.16.x`:
```js
// nuxt.config.js

  build: {
    extend(config) {
      config.resolve.alias['node-fetch-native'] = require.resolve('node-fetch')
    }
  }
 ```

**Comment by KevinBunn** [comment]
I was using D3 with my app and needed a standalone build to import the library. Can confirm setting `build.standalone` in my `nuxt.config` to false makes it so the server finally compiles, but I no longer can import d3. 

I wonder if putting the dependencies in `build.transpile` will allow it to build instead for those who still really need external dependencies as we wait for a solution.

[transpile docs](https://nuxtjs.org/docs/configuration-glossary/configuration-build/#transpile)

**Comment by prinzt** [comment]
> > Any workaround for now?
> 
> Pin your Nuxt version to `"nuxt": "2.15.7"`

Pinning to 2.15.8 seems to work for our team too.

**Comment by sheggi** [comment]
> You could also add a alias from node-fetch-native back to node-fetch when using `v2.16.x`:
> 
> ```js
> // nuxt.config.js
> 
>   build: {
>     extend(config) {
>       config.resolve.alias['node-fetch-native'] = require.resolve('node-fetch')
>     }
>   }
> ```

Worked for me for now.

**Comment by harshadsatra** [comment]
`// nuxt.config.js

  build: {
    extend(config) {
      config.resolve.alias['node-fetch-native'] = require.resolve('node-fetch')
    }
  }`
  
Solution did not work for me, any other suggestions?

**Comment by Jones-S** [comment]
Everything above and including `2.16.0` causes this in my case when combining with `nuxt.config.js` -> `build: { standalone: true }`.
I initially added this because of some ESM problems (https://github.com/nuxt/nuxt/issues/9223) when importing openlayers (https://github.com/openlayers/openlayers).

**Comment by vitalii-bulyzhyn** [comment]
Is there a solution for that? Got the same problem when deploying on Vercel

**Comment by danielroe** [comment]
This should be resolved in the [edge channel](https://nuxt.com/docs/guide/going-further/edge-channel/), or in the next release.

Let me know if not and I'll reopen.

**Comment by danielroe** [comment]
@buglavecz Would you open a new issue with a reproduction? 🙏

**Comment by buglavecz** [comment]
@danielroe yes. I got same error, when I added manually `node-fetch-native` to my project. I will make an issue from this. thanks.

**Comment by buglavecz** [comment]
@danielroe resolved: https://github.com/nuxt/nuxt/issues/19820 😄

**Comment by simonmaass** [comment]
@danielroe I updated to nuxt 2.16.3 with vercel and node 16.x

got the following error: 
<img width="506" alt="image" src="https://user-images.githubusercontent.com/32599710/227214327-efa1c4b1-505a-45c2-8d1e-08113084eed3.png">

**Comment by GelsDEV** [comment]
I had the same problem with Vercel.
Manually adding `node-fetch-native` to the dependencies fixed the problem for me. Also runs flawlessly with Node 18.

**Comment by dommi10** [comment]
I add ``node-fetch-native`` as normal package and it fix the issue

**Comment by joao-conde** [comment]
Adding `node-fetch-native` to my dependencies did not fix the problem for `nuxt@2.17.1`

**Comment by pi0** [comment]
Adding to the dependencies wouldn't probably make a difference (i advice to usually remove such pins and only renew lockfile). 

`node-fetch-native` is bundled with a Syntax that is not supported by Webpack 4. 

Would be nice if you can open a new issue in [nuxt/bridge](https://github.com/nuxt/bridge) to followup any possible solution.
